kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -636,34 +640,125 @@ rightAlignLists internal1 internal2
     (head2, tail2) = Seq.splitAt (length list2 - length list1) list2
 
 matchNormalizedAc
-    :: MonadUnify unifier
-    => (Pair (Builtin.Value normalized (TermLike variable)) -> MatcherT variable unifier ())
+    :: forall normalized unifier variable
+    .   ( Builtin.AcWrapper normalized
+        , MatchingVariable variable
+        , MonadUnify unifier
+        )
+    =>  ( Pair (Builtin.Value normalized (TermLike variable))
+        -> MatcherT variable unifier ()
+        )
     ->  (Builtin.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
         -> TermLike variable
         )
     -> Builtin.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
     -> Builtin.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
     -> MaybeT (MatcherT variable unifier) ()
 matchNormalizedAc pushValue wrapTermLike normalized1 normalized2
-  | [] <- abstract2, [] <- opaque2
-  , [] <- abstract1
+  | [] <- excessAbstract2, [] <- opaque2
```

Comment:
This covers the case when there are no extra abstract elements on either side of the match? It would be good to have some test cases that show when it doesn't match.

Reply:
I changed this condition and added tests.